<a href="https://colab.research.google.com/github/Ammulaanusha2005/AI-ASSISTANT-CODING/blob/main/AI_3_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Prompt Engineering – Improving Prompts and Context Management

1:Zero-Shot Prompting (Palindrome Number Program)

### #Write a Python function to check whether a given number is a palindrome

AI-Genarated code:

In [2]:
n = int(input("Enter a number: "))
s = str(n)

if s == s[::-1]:
    print("Palindrome")
else:
    print("Not a Palindrome")


Enter a number: 123
Not a Palindrome


### Testing Outputs:

input:151->Output:True

input:183->Output:False

Observation: It works for positive numbers and it doesnot handle negative numbers

### 2: One-Shot Prompting (Factorial Calculation)

#Write a Python function to calculate factorial of a number.Example: Input: -4 → Output: Invalid

AI-Generated Code:

In [4]:
def factorial(n):
    if n < 0:
        return "Invalid input"
    fact = 1
    for i in range(1, n + 1):
        fact *= i
    return fact

n = int(input("Enter a number: "))
print(factorial(n))


Enter a number: -4
Invalid input


Testing:

input:3->output:6

input:-4->output:Invalid

Observation: Better clarity than zero-shot method

3:Few-Shot Prompting (Armstrong Number Check)
# Write a Python function to check whether a number is an Armstrong number.

Examples: 371→ Armstrong, 470 → Armstrong, 123 → Not Armstrong

AI-Generated Code:

In [5]:
n = int(input("Enter a number: "))
s = str(n)
total = 0

for i in s:
    total += int(i) ** len(s)

if total == n:
    print("Armstrong Number")
else:
    print("Not an Armstrong Number")


Enter a number: 123
Not an Armstrong Number


Testing:

input: 370->Armstrong Number

input: 123->Not an Armstrong Number

Observation: Few-shot examples improved accuracy and logic of a code

4:Context-Managed Prompting (Optimized Number Classification)
# Write an optimized Python program to classify a number as Prime, Composite, or Neither with input validation.
AI-Generated Code:

Testing:

Input: 7→ Prime

Input: 12 → Composite

Input: 1 → Neither Prime nor Composite

Observation: Optimized and efficient logic.It can solve logical and mathematical solutions

5: Zero-Shot Prompting (Perfect Number Check)
# Write a Python function to check whether a given number is a perfect number
AI-Generated Code:

In [8]:
n = int(input("Enter a number: "))

if n <= 1:
    print("Neither Prime nor Composite")
else:
    for i in range(2, n):
        if n % i == 0:
            print("Composite")
            break
    else:
        print("Prime")
n = int(input("Enter a number: "))
total = 0

for i in range(1, n):
    if n % i == 0:
        total += i

if total == n:
    print("Perfect Number")
else:
    print("Not a Perfect Number")


Enter a number: 33
Composite
Enter a number: 44
Not a Perfect Number


Testing:

6=Perfect Number
18=Not a Perfect Number
97=Not a Perfect Number

Observation: Logic correct but inefficient for large numbers

6: Few-Shot Prompting (Even or Odd Classification with Validation)
# Write a Python program to determine whether a number is even or odd with input validation.
Examples: 12 → Even, 17 → Odd, 0 → Even

AI-Generated Code:

In [6]:
n = int(input("Enter a number: "))

if n % 2 == 0:
    print("Even")
else:
    print("Odd")

Enter a number: 13
Odd


Testing:

Input: 12 → Even

Input: 17 → Odd

Input: -8 → Even

Input: 2.8 → Invalid input

Observation: Few-shot prompting improved validation and clarity